Modules/Core/Common/src/itkObject.cxx

```diff
@@ -503,6 +503,15 @@ Object ::AddObserver(const EventObject & event, Command * cmd) const
   return this->m_SubjectImplementation->AddObserver(event, cmd);
 }
 
+unsigned long
+Object::AddObserver(const EventObject & event, std::function<void(const EventObject &)> function) const
+{
+  auto cmd = FunctionCommand::New();
```

Comment:
Will `cmd` be deleted once object is deallocated?

Reply:
Yes the itk::Object holds onto the Commands with SmartPointers as is normal.